### DISCUSSION TITLE
Why is the `#[inline]` used so frequently?

### DISCUSSION THREAD

**Discussion by jackwener** [paradigmxyz/reth#8105] [General]
I think we don't need to add `#[inline]` for `private function` and `pub(crate) function`, because LLVM generally will inline every private function that is needs inlining.

we just need to mark public functions with #[inline] to enable cross-crate inlining. This will result in bloat if used incorrectly and might result in [slower execution time.](https://stackoverflow.com/questions/49334487/inlining-and-instruction-cache-hit-rates-and-thrashing)

Some material:
- https://matklad.github.io/2021/07/09/inline-in-rust.html
- https://old.reddit.com/r/rust/comments/oh4s2j/blog_post_inline_in_rust/

**Comment by mattsse** [comment]
yeah inline usage could def use a closer look

cc @DaniPopes
